src/test/scala/is/hail/stats/BaldingNicholsModelSuite.scala

```diff
@@ -45,42 +45,64 @@ class BaldingNicholsModelSuite extends SparkSuite {
+      val popArray = bnm.sampleAnnotations.toArray.map(_.asInstanceOf[GenericRow](0).asInstanceOf[GenericRow](0).toString.toDouble)
+      val popCounts = popArray.groupBy(x => x).values.toSeq.sortBy(_(0)).map(_.size)
+
+      popCounts.indices.foreach(index => {
+        assertEquals(popCounts(index) / N, popDist(index), Math.ceil(N / K * .1))
+      })
 
-    val popDist: Array[Double] = popDistOpt.getOrElse(Array.fill(K)(1.0))
+      //Test AF distributions
+      val arrayOfVARows = bnm.variantsAndAnnotations.collect().map(_._2).toSeq.asInstanceOf[mutable.WrappedArray[GenericRow]]
+      val arrayOfVATuples = arrayOfVARows.map(row => row.get(0).asInstanceOf[GenericRow]).map(row => (row.get(0), row.get(1)).asInstanceOf[(Double, Vector[Double])])
 
-    //Test population distribution
-    val popArray = bnm.sampleAnnotations.toArray.map(_.asInstanceOf[GenericRow](0).asInstanceOf[GenericRow](0).toString.toDouble)
-    val popCounts = popArray.groupBy(x => x).values.toSeq.sortBy(_(0)).map(_.size)
+      val AFStats = arrayOfVATuples.map(tuple => meanAndVariance(tuple._2))
 
-    popCounts.indices.foreach(index => {
-      assertEquals(popCounts(index) / N, popDist(index), Math.ceil(N / K * .1))
-    })
+      arrayOfVATuples.map(_._1).zip(AFStats).foreach{
+        case (p, mv) =>
+          assertEquals(p, mv.mean, .2) //Consider alternatives to .2
+          assertEquals(.1 * p * (1 - p), mv.variance, .1)
+      }
 
-    //Test AF distributions
-    val arrayOfVARows = bnm.variantsAndAnnotations.collect().map(_._2).toSeq.asInstanceOf[mutable.WrappedArray[GenericRow]]
-    val arrayOfVATuples = arrayOfVARows.map(row => row.get(0).asInstanceOf[GenericRow]).map(row => (row.get(0), row.get(1)).asInstanceOf[(Double, Vector[Double])])
+      //Test genotype distributions
+      val meanGeno_mk = bnm.rdd
+        .map(_._2._2.zip(popArray).groupBy(_._2).toSeq.sortBy(_._1))
+        .map(_.map(popIterPair => mean(popIterPair._2.map(_._1.gt.get.toDouble))).toArray)
+        .collect()
 
-    val AFStats = arrayOfVATuples.map(tuple => meanAndVariance(tuple._2))
+      val p_mk = arrayOfVATuples.map(_._2.toArray)
 
-    arrayOfVATuples.map(_._1).zip(AFStats).foreach{
-      case (p, mv) =>
-        assertEquals(p, mv.mean, .2) //Consider alternatives to .2
-        assertEquals(.1 * p * (1 - p), mv.variance, .1)
+      val meanDiff = (meanGeno_mk.flatten.sum - 2 * p_mk.flatten.sum) / (M * K)
+      assertEquals(meanDiff, 0, .1)
     }
+  }
+
+
 
-    //Test genotype distributions
-    val meanGeno_mk = bnm.rdd
-      .map(_._2._2.zip(popArray).groupBy(_._2).toSeq.sortBy(_._1))
-      .map(_.map(popIterPair => mean(popIterPair._2.map(_._1.gt.get.toDouble))).toArray)
-      .collect()
+  @Test def testAFRanges() {
+    testRangeHelp(TruncatedBetaDist(.01, 2, .2, .8), .2, .8, 17)
+    testRangeHelp(TruncatedBetaDist(3, 3, .4, .6), .4, .6, 12)
 
-    val p_mk = arrayOfVATuples.map(_._2.toArray)
+    testRangeHelp(UniformDist(.4, .7), .4, .7, 9823)
```

Comment:
use small seeds in these like 0, 1, 2, 3.  otherwise i'm left wondering what is special about 9823